Makefile.base

```diff
@@ -65,10 +65,17 @@ DEP := $(OBJC:.o=.d) $(OBJCXX:.o=.d) $(ASSMOBJ:.o=.d)
 $(BINDIR)/$(MODULE)/:
 	$(Q)mkdir -p $@
 
-$(BINDIR)/$(MODULE).a $(OBJ): | $(BINDIR)/$(MODULE)/
+$(BINDIR)/$(MODULE).o $(OBJ): | $(BINDIR)/$(MODULE)/
 
-$(BINDIR)/$(MODULE).a: $(OBJ) | ${DIRS:%=ALL--%}
-	$(Q)$(AR) $(ARFLAGS) $@ $?
+# An empty object file is included in the link to avoid situations where modules
+# do not provide any code (e.g. sys, some boards), causing the build to fail
+# with a missing object file.
+EMPTYOBJ = $(BINDIR)/$(MODULE)/.empty.o
```

Comment:
I am not sure that using a generic empty.o will work. If I'm not mistaken (I did not test it), the linker will reject an object file if its not the right architecture.

I guess something like that would be more robust:

```make
ifeq $(,$(OBJ))
  OBJ := $(BINDIR)/$(MODULE)/empty.o

  $(BINDIR)/$(MODULE)/empty.o: $(RIOTBASE)/empty.c
        $(CC) $(CFLAGS) -w -c -o $@ $<
ifeq
```

```c
static void __attribute__((unused)) _placeholder(void) {}
```

Reply:
It works, try it. It is the correct architecture because of using the real CFLAGS when generating it.
Not sure the ifeq suggestion is robust because of lazy evaluation of recursive variables, does OBJ always have its final value when interpreting this makefile?